### PR TITLE
[2697] [2709] [2698] [2699] Primary and secondary course details flow

### PR DESCRIPTION
**Pull request by archferns**
### Context
Following https://github.com/DFE-Digital/register-trainee-teachers/pull/1425, split out the primary/secondary course details forms.

### Changes proposed in this pull request
1. Add a new field `primary_course_subjects` which could take one of the specified primary subject combos.
2. Split out the age ranges for primary or secondary phases.
3. Subject picker uses a new scope `SubjectSpecialism.secondary` to remove 3 primary subjects, namely: early years teaching, primary teaching, specialist teaching (primary with mathematics) 

### Guidance to review
1. Visit the "Course Details" against a trainee, select Primary or secondary when asked "Which phase of education will the trainee teach?" and play around with the following forms.

### CODE REVIEW

**Comment by archferns** [comment]
[[L] Primary course details flow](https://trello.com/c/NmhmeaFm/2698-l-primary-course-details-flow)

**Comment by archferns** [comment]
[[M] Secondary course details flow](https://trello.com/c/mw2lBdwn/2699-m-secondary-course-details-flow)

**Review by darokel — COMMENTED** [review]
Code wise looking good. I found some snags. Let me know how you want to handle them. Either we can do these in a sep ticket or maybe you could pick off a few of them in this PR if they're related?:

1. Selecting primary, defaults the option to "primary with another subject"

Screenshot 2021-09-15 at 17 24 46


2. Education phase on confirm page should be Capitilised

Screenshot 2021-09-15 at 17 26 06

3. Clicking the education phase change link on the confirm page takes you back to the primary form where it should take you to the initial flow to change between primary/secondary education phases

4. Something to confirm with Ed, we show the display names on the primary course form should we also show it on the confirm page? At the moment we show the underlying subjects they map to but the prototype still shows the display name

5. Small edge case. If you select "Primary with science" then go to change to "Primary with another subject" the second subject is pre-populated with "science" but is not actually listed in the picker as an option so the dropdown value displays "undefined"

Screenshot 2021-09-15 at 17 32 34

File: config/initializers/mappings/publish_to_allocation_subject_mapping.rb [hunk 1]
```diff
@@ -28,6 +46,17 @@ module PublishSubjects
...
 
+PUBLISH_PRIMARY_SUBJECT_SPECIALISM_MAPPING = {
+  # Primary subject and its derivatives
+  PublishSubjects::PRIMARY => [CourseSubjects::PRIMARY_TEACHING],
+  PublishSubjects::PRIMARY_WITH_ENGLISH => [CourseSubjects::PRIMARY_TEACHING, CourseSubjects::ENGLISH_STUDIES],
+  PublishSubjects::PRIMARY_WITH_GEOGRAPHY_AND_HISTORY => [CourseSubjects::PRIMARY_TEACHING, CourseSubjects::GEOGRAPHY, CourseSubjects::HISTORY],
+  PublishSubjects::PRIMARY_WITH_MATHEMATICS => [CourseSubjects::SPECIALIST_TEACHING_PRIMARY_WITH_MATHEMETICS],
+  PublishSubjects::PRIMARY_WITH_MODERN_LANGUAGES => [CourseSubjects::PRIMARY_TEACHING, CourseSubjects::MODERN_LANGUAGES],
+  PublishSubjects::PRIMARY_WITH_PHYSICAL_EDUCATION => [CourseSubjects::PRIMARY_TEACHING, CourseSubjects::PHYSICAL_EDUCATION],
+  PublishSubjects::PRIMARY_WITH_SCIENCE => [CourseSubjects::PRIMARY_TEACHING, CourseSubjects::GENERAL_SCIENCES],
```
Comment (archferns) [inline config/initializers/mappings/publish_to_allocation_subject_mapping.rb:57]: Note that the subject has changed from 
`[CourseSubjects::PRIMARY_TEACHING, CourseSubjects::SCIENCE]`
`[CourseSubjects::PRIMARY_TEACHING, CourseSubjects::GENERAL_SCIENCES]`
following https://github.com/DFE-Digital/register-trainee-teachers/pull/1414/commits/49e545e3bc109161f15efe431cb07e23b6989a1f on https://github.com/DFE-Digital/register-trainee-teachers/pull/1414

**Review by darokel — APPROVED** [review]

**Pull request merged**